compute/disk-images/create_disk_image_from_snapshot.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_images_create_from_snapshot]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "cloud.google.com/go/compute/apiv1/computepb"
+)
+
+// Creates a disk image from an existing image
+func createImageFromSnapshot(
+	w io.Writer,
+	projectID, snapshotName, imageName string,
+	storageLocations []string,
```

Comment:
do we need to include `storageLocations` and `guestOSFeatures` as parameters here? what value are they adding to the sample? It would be easier to inline their use in the function - these functions dont need to be general purpose.